metadata_backend/api/handlers/object.py

```diff
@@ -120,6 +120,15 @@ async def post_object(self, req: Request) -> Response:
         db_client = req.app["db_client"]
         submission_op = SubmissionOperator(db_client)
 
+        # check if submission is already published
+        # objects shouldn't be added to published submissions
+        try:
+            await submission_op.check_submission_published(submission_id)
+        except web.HTTPException as exc:
+            reason = f"Submission with id {submission_id} is published and more objects cannot be added to it."
+            LOG.error(reason)
+            raise web.HTTPUnauthorized(reason=reason) from exc
+
```

Comment:
i think a simple `await submission_op.check_submission_published(submission_id)` would do as the `check_submission_published` raises `HTTPUnauthorized`

Reply:
Yea, it just gives a different reason with the `HTTPException`. But I was also thinking of just altering the `check_submission_published` reason string to something more general